tags/middleware.py

```diff
@@ -0,0 +1,49 @@
+from django.db.models import Q
+from tags.models import Tag
+
+
+class InjectTagFilteringMiddleware:
```

Comment:
Explain what it does

Reply:
It's a middle ware that injects a tag filter 😇